### ISSUE TITLE
Incorrect playground endpoint in apollo-server-lambda

### ISSUE THREAD

**Issue by danielpatrickdotdev** [apollographql/apollo-server#4687]
When deploying `apollo-server-lambda` as per the [readme](https://github.com/apollographql/apollo-server/blob/main/packages/apollo-server-lambda/README.md), the url used by the playground is wrong when using the API gateway execute-api domain.

I believe this is caused by the change made in #1527.

This [example payload](https://github.com/awsdocs/aws-lambda-developer-guide/blob/master/sample-apps/nodejs-apig/event.json) for this type of event includes a url of `https://70ixmpl4fl.execute-api.ca-central-1.amazonaws.com/Prod/`, with `event.requestContext.path` of `/Prod/` (the stage name). The `event.path` is `/`. Clearly in this case using `event.path` would be incorrect.

When using an API gateway with a custom domain, the values for `event.requestContext.path` and `event.path` are identical.

I'd be happy to submit a PR to fix this, but it would be good to understand why this was changed in the first place first as I don't want to break anything.

**Comment by arabold** [comment]
I'm having the same issue. `apollo-server-lambda` only works with custom domains but not in a simple default setup as it is described in the readme. The original "fix" you're referring to is nearly two years old. That makes me wonder why hasn't this been noted earlier?! We can't be the only ones running into this, can we?

**Comment by glasser** [comment]
Is this related to #4891/#4892? That's another issue about path handling in Lambda. Oddly that one suggests that it only works *without* custom domains, whereas it sounds like you're saying playground only works *with* custom domains?

I'm happy to merge a fix and have #4892 already lined up for my next PRs-outside-my-current-knowledge-learning-and-merging day); maybe #4892 needs to be expanded to fix this too?

**Comment by arabold** [comment]
I did some digging and more testing... The root cause might lie in `serverless-offline`: The default setting appends the stage name to the URL path, e.g. `http://localhost:3000/dev/` for the `dev` stage. I found a related issue over at their repository: https://github.com/dherault/serverless-offline/issues/1029

It seems that the fix in #1527 trips over this behavior/bug in `serverless-offline`.

For now I have set up my local `serverless-offline` with the `noPrependStageInUrl: true` option to avoid the stage name in my path. This matches my actual AWS Lambda setup in which the stage name is defined in the domain mapping (`dev.example.com`). This works as expected now.

**tl;dr.** I think this issue can be closed with a reference to the previously mentioned bug report for `serverless-offline`: https://github.com/dherault/serverless-offline/issues/1029. As a workaround set `noPrependStageInUrl: true` in your `serverless.yml`.

**Comment by remarkablerocket** [comment]
I didn't use the Serverless framework (I used SAM/CloudFormation) so unsure how this could be related?

@glasser  without looking into it in too much detail, I don't _think_ these issues are related but unsure as I've not played around with non-default base path mapping.

**Comment by glasser** [comment]
While it's possible that the original author of #1527 (@stubailo) or its reviewer (@jbaxleyiii) might choose to chime in, neither of them work at Apollo any more, and it's quite likely that the world of Lambda has changed a lot in the past 3 years anyway.

I've merged the other semi-related PR (#4892) but I'm not sure I understand Lambda well enough to get what's going on here.

That said:
- If you send a PR that fixes this, I'll definitely review it, and will do my best to gather info during a release process from Lambda folks to make sure this doesn't break others
- We are likely to change the graphql-playground integration in AS3 as part of the goal of that repo is reducing the number of direct dependencies that AS has on other packages which could be separately integrated.

**Comment by glasser** [comment]
Apollo Server 4 replaces a hard-coded set of web framework integrations with a simple stable API for building integrations. As part of this, the core project no longer directly supports a Lambda integration. Check out the [`@as-integrations/aws-lambda`](https://www.npmjs.com/package/@as-integrations/aws-lambda) package; this new package is maintained by Lambda users and implements support for the API Gateway v1 and v2 protocols. If you need to work with different AWS projects, you may be able to add that support to that package, or take another approach such as combining the [`@vendia/serverless-express`](https://www.npmjs.com/package/@vendia/serverless-express) package with Apollo Server 4's built-in `expressMiddleware` (this is the same approach that `apollo-server-lambda` takes in AS3). Sorry for the delay!

(In addition, we changed the support for Playground only to supporting multiple landing pages in AS3 without parsing paths, and AS4 doesn't parse paths at all.)
